src/Lean/Meta/Basic.lean

```diff
@@ -883,6 +883,29 @@ def withLocalDecl (name : Name) (bi : BinderInfo) (type : Expr) (k : Expr → n
 def withLocalDeclD (name : Name) (type : Expr) (k : Expr → n α) : n α :=
   withLocalDecl name BinderInfo.default type k
 
+partial def withLocalDecls 
+  [Inhabited α]
+  (declInfos : Array (Name × BinderInfo × (Array Expr → Expr)))
```

Comment:
I am wondering whether we need `Array Expr -> MetaM Expr`.

Reply:
I did it with `Array Expr -> n Expr` like the rest of the function.